lib/internal/http2.js

```diff
@@ -48,7 +46,139 @@ Object.defineProperty(exports, 'constants', {
 // emit events as callbacks. To allow this to happen,
 // it has to inherit from EventEmitter.
 util.inherits(http2.Http2Session, EventEmitter);
+util.inherits(http2.Http2Stream, Duplex);
 
+function maybeDestroyStream(stream) {
+  if ((!stream.readable &&
+       !stream.writable &&
+       !stream._writableState.length)) {
+    stream.destroy();
+  }
```

Comment:
the best check here is checking if `destroy` exists. There is no guarantee, as its not part of the API.
Ideally we can do `stream.destroy(err)`.